### PR TITLE
[language server / incrmental builder] allow reusing resource set

### PR DESCRIPTION
**Pull request by svenefftinge**

### CODE REVIEW

**Comment by svenefftinge** [comment]
This is for post 2.11

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/ProjectManager.xtend [hunk 1]
```diff
@@ -105,17 +105,26 @@ class ProjectManager {
...
+        } else {
+            val resDescs = ChunkedResourceDescriptions.findInEmfObject(this.resourceSet);
+            resDescs.setContainer(projectDescription.name, newIndex)   
+        }
+        return this.resourceSet;
     }
     
     def Resource getResource(URI uri) {
-        resourceSet.getResource(uri, true)
+        val resource = resourceSet.getResource(uri, true)
```
Comment (akosyakov) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/ProjectManager.xtend]: maybe we should pass `false` here and check whether the result is null or not?

Comment (svenefftinge) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/ProjectManager.xtend]: Loading it should be ok. We just need to make sure that it is not done in parallel.
I.e. we must not allow parallel read access if we are not sure that everything has been fully initialized.

**Comment by spoenemann** [comment]
There are failing tests:
http://services.typefox.io/open-source/jenkins/job/xtext-core/job/master/320/

**Comment by svenefftinge** [comment]
I am looking into it

**Pull request merged**